terraform/cache/config.go

```diff
@@ -0,0 +1,100 @@
+package cache
+
+import (
+	"net"
+	"strconv"
+	"time"
+)
+
+const (
+	defaultHostname        = "localhost"
+	defaultShutdownTimeout = time.Second * 30
+)
+
+type Option func(Config) Config
+
+func WithHostname(hostname string) Option {
```

Comment:
Do we have a test that checks this server can't be accessed without proper auth?